### DISCUSSION TITLE
Createdb fails to create clickhouse database

### DISCUSSION THREAD

**Discussion by cynthiahenaff** [plausible/analytics#1433] [Self-Hosted Support]
### Past Issues Searched

- [X] I have searched open and closed issues to make sure that the bug has not yet been reported

### Issue is a Bug Report

- [X] This is a bug report and not a feature request, nor asking for self-hosted support

### Describe the bug

Hi,

I'm trying to start Plausible on my own with docker-compose.

On createdb, Clickhouse is running, the createdb script connects to it and triggers errors about plausible_events_db database not existing.
AFAIK, these errors are normal.

But after that I get an error "no match of right hand side value" that seems to be a timeout issue if I understand it well.

It seems that `queue_target` and `queue_interval` are defined in https://github.com/plausible/analytics/blob/master/config/runtime.exs#L227

I'm not sure if it is my clickhouse that is not responding as fast as it should or anything else.
Do you have any idea?

```
~ $ /entrypoint.sh db createdb
Loading plausible..
Starting dependencies..
Starting repos..
create Plausible.Repo database if it doesn't exist

17:39:17.548 [error] Clickhousex.Protocol (#PID<0.179.0>) failed to connect: ** (ErlangError) Erlang error: "Code: 81, e.displayText() = DB::Exception: Database plausible_events_db doesn't exist (version 21.3.2.5 (official build))\n"

17:39:17.548 [error] Clickhousex.Protocol (#PID<0.178.0>) failed to connect: ** (ErlangError) Erlang error: "Code: 81, e.displayText() = DB::Exception: Database plausible_events_db doesn't exist (version 21.3.2.5 (official build))\n"

17:39:20.093 [error] Clickhousex.Protocol (#PID<0.179.0>) failed to connect: ** (ErlangError) Erlang error: "Code: 81, e.displayText() = DB::Exception: Database plausible_events_db doesn't exist (version 21.3.2.5 (official build))\n"

17:39:20.382 [error] Clickhousex.Protocol (#PID<0.178.0>) failed to connect: ** (ErlangError) Erlang error: "Code: 81, e.displayText() = DB::Exception: Database plausible_events_db doesn't exist (version 21.3.2.5 (official build))\n"

** (MatchError) no match of right hand side value: {:error, "connection not available and request was dropped from queue after 2992ms. This means requests are coming in and your connection pool cannot serve them fast enough. You can address this by:\n\n  1. By tracking down slow queries and making sure they are running fast enough\n  2. Increasing the pool_size (albeit it increases resource consumption)\n  3. Allow requests to wait longer by increasing :queue_target and :queue_interval\n\nSee DBConnection.start_link/2 for more information\n"}
    (plausible 0.0.1) lib/plausible_release.ex:48: anonymous fn/2 in Plausible.Release.createdb/0
    (elixir 1.11.3) lib/enum.ex:2193: Enum."-reduce/3-lists^foldl/2-0-"/3
    (plausible 0.0.1) lib/plausible_release.ex:47: Plausible.Release.createdb/0
    (stdlib 3.14) erl_eval.erl:680: :erl_eval.do_apply/6
    (elixir 1.11.3) lib/code.ex:341: Code.eval_string_with_error_handling/3

~ $ echo $?
1
```

### Expected behavior

Creation of plausible_events_db

### Environment

```markdown
- OS:
- Browser:
- Browser Version:
```

**Comment by StoryHack** [comment]
I am having the same error when I try to run plausible behind a reverse proxy.

What changes, if any, have you made to the docker-compose.yml? I can get the service to boot on a testing machine locally, and hit the unencrypted site on localhost:8000 just using the provided .yml & filled out .env file.

However, when I add the stuff to the compose .yml file so that it should work behind the reverse proxy, the plausible server container starts barfing the error the OP pasted and restarting ad nauseum. Both db containers and the mail container stay up and stable.

I'm using caddy-docker-proxy as the reverse proxy, and have it working for other containers, so I'm fairly confident that I have that set up correctly.

I have tried it both on a digital ocean droplet and on a local machine running docker, with the same results.

**Comment by productdevbook** [comment]
some problem 1.4.4
